ITISFoundation/osparc-simcore
# ♻️🔨  ``catalog``: fixes mypy issues and add type-check step in CI

## What do these changes do?

Fixes mypy errors in ``catalog``  and adds ``type-check`` step in CI

## Related issue/s


- related #4071


## How to test

```cmd
cd services/catalog
make install-dev
make mypy
```
should not show any error
